### PR TITLE
Add explorer context menus for 'Run/Debug Pester tests'

### PR DESCRIPTION
**Pull request by bergmeister**
## PR Summary

Similar to the existing context menu for the files tab. Manually tested that it works

image

cc @JustinGrote @nohwnd

## PR Checklist

Note: Tick the boxes below that apply to this pull request by putting an `x` between the square brackets.
Please mark anything not applicable to this PR `NA`.

- [x] PR has a meaningful title
- [x] Summarized changes
- [ ] PR has tests
- [x] This PR is ready to merge and is not work in progress
    - If the PR is work in progress, please add the prefix `WIP:` to the beginning of the title and remove the prefix when the PR is ready

### CODE REVIEW

**Comment by bergmeister** [comment]
@poshchan restart ci please

**Comment by PoshChan** [comment]
@bergmeister, all requests start with the magic word: `Please`

<details>
<summary>Commands available in this repo for you:
</summary>
<ul><li>
<code>get failures</code> this will attempt to get the latest failures for all of the target pipelines
</li>
<li>
<code>remind me in &lt;value&gt; &lt;units&gt;</code> this will create a reminder that will be posted after the specified duration &lt;value&gt; is a number, and &lt;units&gt; can be <code>minutes</code>, <code>hours</code>, or <code>days</code> (singular or plural)
</li>

</ul>
</details>

**Comment by bergmeister** [comment]
@poshchan please restart ci!

**Comment by PoshChan** [comment]
@bergmeister, you are not authorized to request a rebuild

**Comment by TylerLeonhardt** [comment]
you can ignore the macos failure

**Comment by bergmeister** [comment]
@TylerLeonhardt Just something that is worthwhile noting is that it depends on the currently displayed and focused file in the editor, i.e. one needs to click on the item in explorer first and then do the right click. If a different file has the focus then that file would be chosen for the test run. Also, if someone never opened any ps1 file, then it would just silently fail because the PSIC is not up and running.

**Comment by nohwnd** [comment]
Hmm that’s a bit unfortunate, I know I never left-click, right-click, I just right-click.

**Comment by bergmeister** [comment]
Currently we get the path from `const uriString = vscode.window.activeTextEditor.document.uri.toString();` in all use cases (file tab and explorer context menu). Maybe we need to enhance to check if we were called from explorer or from the file tab menu (or pass a parameter to the method with that info) so that we can use a more appropriate property for the file explorer use case.

**Comment by bergmeister** [comment]
I could fix it to pass the `fileUri` of the right-clicked explorer menu through (which works even if that explorer item is not selected) and tested the file tab context menu still works as well. Therefore this issue is gone.

File: src/features/PesterTests.ts [hunk 1]
```diff
@@ -51,8 +51,8 @@ export class PesterTestsFeature implements IFeature {
         this.languageClient = languageClient;
     }
 
-    private launchAllTestsInActiveEditor(launchType: LaunchType) {
-        const uriString = vscode.window.activeTextEditor.document.uri.toString();
+    private launchAllTestsInActiveEditor(launchType: LaunchType, fileUri?: vscode.Uri) {
```
Comment (TylerLeonhardt) [inline src/features/PesterTests.ts]: you made this optional but then use `fileUri.toString()`. Can you either make this not optional or null check the `fileUri` in the method?

```
fileUri ? fileUri.toString() : vscode.window.activeTextEditor.document.uri.toString()
```

Comment (bergmeister) [inline src/features/PesterTests.ts]:
```suggestion
    private launchAllTestsInActiveEditor(launchType: LaunchType, fileUri: vscode.Uri) {
```

Comment (bergmeister) [reply src/features/PesterTests.ts]: Thanks for spotting it. This was a left-over from development when I was trying various approaches. Cleaned it up. `fileUri` is not optional any more now

**Pull request merged**